network/onchain-discovery/src/client.rs

```diff
@@ -293,6 +279,11 @@ where
                     new_state.latest_version(),
                 );
                 self.trusted_state = new_state;
+
+                let discovery_set = opt_discovery_set.expect(
+                    "We're guaranteed Some(DiscoverySet) from QueryDiscoverySetResponse::verify_and_ratchet"
+                );
```

Comment:
if we are guaranteed, then the API should return a discovery_set and not an optional one, no?

Reply:
ahh only guaranteed Some if it's there's a non-empty epoch change proof. that's why it's only in the `TrustedStateChange::Epoch` branch